components/payment.js

```diff
@@ -105,7 +123,7 @@ const useInvoice = () => {
     return inv
   }, [cancelInvoice])
 
-  return { create, isPaid, waitUntilPaid, cancel }
+  return { create, isPaid, waitUntilPaid: waitController.wait, stopWaiting: waitController.stop, cancel }
```

Comment:
Maybe it makes sense to rename it to `pollUntilPaid` and `stopPolling`.

Reply:
hmmm I like your original `wait` names though because poll doesn't imply blocking.